src/networkaddress.rs

```diff
@@ -59,17 +59,22 @@ impl NetworkAddress for SocketAddrV4 {
                 >= std::mem::align_of::<libc::sockaddr_in>()
         );
 
-        let mut result = zeroed_sockaddr_storage();
-        // Safety: the above assertions guarantee that alignment and size are correct.
-        // the resulting reference won't outlast the function, and result lives the entire
-        // duration of the function
-        let out = unsafe { &mut (*(&mut result as *mut _ as *mut libc::sockaddr_in)) };
-        out.sin_family = libc::AF_INET as _;
-        out.sin_port = u16::from_ne_bytes(self.port().to_be_bytes());
-        out.sin_addr = libc::in_addr {
-            s_addr: u32::from_ne_bytes(self.ip().octets()),
+        let sockaddr = libc::sockaddr_in {
+            sin_family: libc::AF_INET as _,
+            sin_port: u16::from_ne_bytes(self.port().to_be_bytes()),
+            sin_addr: libc::in_addr {
+                s_addr: u32::from_ne_bytes(self.ip().octets()),
+            },
+            sin_zero: [0; 8],
         };
```

Comment:
show clearly that the whole struct is initialized

Reply:
Ok, for me this feels weird because what we want is not a sockaddr_in. We want a sockaddr(_storage) containing an ipv4 address, which for me the original formulation captures better.